modules/generate-reference-doc/src/main/scala/scala/cli/doc/GenerateReferenceDoc.scala

```diff
@@ -293,16 +422,46 @@ object GenerateReferenceDoc extends CaseApp[InternalDocOptions] {
         }
       }
 
-    addHandlers(usingHandlers)
+    if (onlyRestricted) {
+      // TODO add links to RFC
+      b.section(
+        "This document is a specification of `scala` runner.",
+        "For now it uses documentation specific to Scala CLI but at some point it may be refactor to provide more abstract documentation.",
```

Comment:
```suggestion
        "For now it uses documentation specific to Scala CLI but at some point it may be refactored to provide more abstract documentation.",
```